spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/web/servlet/WebMvcMetricsConfiguration.java

```diff
@@ -61,27 +60,7 @@ public WebMvcMetrics controllerMetrics(MeterRegistry registry,
 	}
 
 	@Bean
-	public MetricsHandlerInterceptor webMetricsInterceptor(
-			WebMvcMetrics controllerMetrics) {
-		return new MetricsHandlerInterceptor(controllerMetrics);
+	public MetricsFilter webMetricsFilter(WebMvcMetrics controllerMetrics, HandlerMappingIntrospector introspector) {
```

Comment:
Given that this is now a filter, it feels odd that it’s in Spring MVC-specific auto-configuration

Reply:
How do you suggest conditionalizing it to servlet-based APIs?